### PR TITLE
Add msgstream seek

### PR DESCRIPTION
**Pull request by godchen0212**
Add msgstream seek.

Signed-off-by: godchen <qingxiang.chen@zilliz.com>

### CODE REVIEW

File: internal/msgstream/result [hunk 1]
```diff
@@ -0,0 +1,63 @@
+=== RUN   TestStream_PulsarMsgStream_3
```
Comment (neza2017) [inline internal/msgstream/result:1]: what does this file do?

Comment (godchen0212) [reply internal/msgstream/result:1]: It is used to test mqMsgStream's seek function(different from mqTtMsgStream).

Comment (godchen0212) [reply internal/msgstream/result:1]: I have made changes in this test's name.

File: internal/msgstream/mq_msgstream.go [hunk 2]
```diff
@@ -351,7 +354,17 @@ func (ms *mqMsgStream) receiveMsg(consumer mqclient.Consumer) {
 				tsMsg.SetTraceCtx(opentracing.ContextWithSpan(context.Background(), sp))
 			}
 
-			msgPack := MsgPack{Msgs: []TsMsg{tsMsg}}
+			tsMsg.SetPosition(&MsgPosition{
```
Comment (xige-16) [inline internal/msgstream/mq_msgstream.go:357]: The message position has been set by getTsMsgFromConsumerMsg

File: internal/msgstream/mq_msgstream.go [hunk 3]
```diff
@@ -128,8 +129,6 @@ func (ms *mqMsgStream) AsConsumer(channels []string, subName string) {
 			ms.consumers[channel] = pc
 			ms.consumerChannels = append(ms.consumerChannels, channel)
 			ms.consumerLock.Unlock()
-			ms.wait.Add(1)
```
Comment (xige-16) [inline internal/msgstream/mq_msgstream.go:131]: This way, you can’t join the consumer after start msgstream

**Review by xige-16 — APPROVED** [review]

**Pull request merged**